rust/worker/src/execution/operator.rs

```diff
@@ -66,18 +100,58 @@ pub(crate) trait TaskWrapper: Send + Debug {
 #[async_trait]
 impl<Input, Output, Error> TaskWrapper for Task<Input, Output, Error>
 where
-    Error: Debug,
+    Error: Debug + Send,
     Input: Send + Sync + Debug,
     Output: Send + Sync + Debug,
 {
     async fn run(&self) {
-        let result = self.operator.run(&self.input).await;
-        let task_result = TaskResult {
-            result,
-            task_id: self.task_id,
+        let result = AssertUnwindSafe(self.operator.run(&self.input))
+            .catch_unwind()
+            .await;
+
+        match result {
+            Ok(result) => {
+                // If this (or similarily, the .send() below) errors, it means the receiver was dropped.
+                // There are valid reasons for this to happen (e.g. the component was stopped) so we ignore the error.
```

Comment:
review: I believe this reasoning is correct, but want to double check
previously, any errors were silently ignored, but that behavior was implicit

Reply:
i think we should just log an error if that happens, since in our code, we don't expect it anywhere by construction but its technically possible

Reply:
couldn't it happen during component shutdown like commented?

Reply:
yes, my point is we only are in that case if the server is shutting down. We don't have any code paths that we'ed expect this to be occurring I think?

Reply:
right, but shutting down is part of the normal lifecycle
I'll add logging for now and we can always adjust later :)

Reply:
b430a5ce